utils/index.ts

```diff
@@ -1,3 +1,4 @@
 export * from "./utils";
+export * from "./utils.hre";
```

Comment:
Exporting this directly into the index.ts means that _every importer_ will have to have a hardhat config and will automatically start hardhat in process. This is a bad pattern for a package. Hardhat shouldn't be used this way.

What in the relayer repo depends on this? This file exports deployNewProxy and the hre object as far as I can see, and I can't find deployNewProxy anywhere in the relayer repo. Are we sure the relayer-v2 repo isn't just importing `hre` from here instead of just importing it on its own: `import hre from "hardhat";`?

Side note: IMO we should also just avoid re-exporting hre in the future and just pull it in directly via importing hardhat wherever needed. Re-exports are an antipattern unless the dependency needs to be modified or initialized. I think that's how we got here, and I think we just did that for "cleanliness" so we could condense imports (which doesn't seem worthwhile in hindsight.

Reply:
Understood - makes sense. I was missing some of the context on how hardhat is intended to be used and had assumed to re-exported from contracts-v2 for a reason.

It works fine to import hardhat directly in the places that need it within relayer-v2.